activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -636,7 +636,17 @@ def disable_referential_integrity
 
       # Override to check all foreign key constraints in a database.
       def all_foreign_keys_valid?
+        check_all_foreign_keys_valid!
         true
+      rescue ActiveRecord::StatementInvalid
+        false
+      end
+      deprecate :all_foreign_keys_valid?, deprecator: ActiveRecord.deprecator
+
+      # Override to check all foreign key constraints in a database.
+      # The adapter should raise a +ActiveRecord::StatementInvalid+ if foreign key
+      # constraints are not met.
+      def check_all_foreign_keys_valid!
       end
```

Comment:
I still believe we should be calling the old method in the case that it is overridden in the subclass, otherwise this would constitute a breaking change for existing 3rd party adapters that have not implemented the new method yet. I know this is unfavorable due to the use of reflection, but I believe it is the correct course of action:
```suggestion
      def check_all_foreign_keys_valid!
        return if self.class.method(:all_foreign_keys_valid?).owner == AbstractAdapter

        raise unless all_foreign_keys_valid?
      end
```

Reply:
Hmm, I think @yahonda 's argument [here](https://github.com/rails/rails/pull/44943#discussion_r919848770) was that we should just alert 3rd party maintainers (of which he is one).

Based on a brief github search I found https://github.com/jruby/activerecord-jdbc-adapter/blob/e96af8f2d0fb5960a93b5a2e6f01e7f3809ac6cf/lib/arjdbc/sqlite3/adapter.rb#L221 and yahonda's oracle adapter.